### PR TITLE
tx history new ui

### PR DESCRIPTION
**Pull request by OrenZak**
#### Main purpose:
Create new UI for order history page
#### Technical description:
- Split the orders into two recyclers 
- Created `KinEcosysemTabs` custom view
- Handled all the transitions

### CODE REVIEW

File: sdk/src/main/java/com/kin/ecosystem/history/presenter/OrderHistoryPresenter.kt [hunk 1]
```diff
@@ -0,0 +1,282 @@
...
+                    addOrder(order = order)
+                } else {
+                    //Update
+                    updateOrder(index = index, order = order)
+                }
+            }
+        }
+    }
+
+    private fun splitByType(list: List<Order>, earnList: MutableList<Order>, spendList: MutableList<Order>) {
```
Comment (yosriz) [inline sdk/src/main/java/com/kin/ecosystem/history/presenter/OrderHistoryPresenter.kt]: warning! Kotlin show off:
```
    private fun splitByType(list: List<Order>) : Pair<List<Order>, List<Order>>{
        return list.filter { it.status != Status.PENDING }
                .partition { isEarn(it) }
    }
```
and then in `syncNewOrders`:
```
   private fun syncNewOrders(newOrdersListObj: OrderList) {
        val (newEarnList,newSpendList ) = splitByType(newOrdersListObj.orders)
        addOrders(newEarnList, earnOrderList)
        addOrders(newSpendList, spendOrderList)
    }
```

File: sdk/src/main/java/com/kin/ecosystem/history/presenter/OrderHistoryPresenter.kt [hunk 2]
```diff
@@ -0,0 +1,282 @@
...
+        }
+        orderRepository.addOrderObserver(completedOrderObserver!!)
+    }
+
+    private fun isCurrentOrder(order: Order): Boolean {
+        return currentPendingOrder?.let { it == order } ?: false
+    }
+
+    private fun getStatus(@NonNull order: Order): OrderStatus {
+        return when (order.status!!) {
```
Comment (yosriz) [inline sdk/src/main/java/com/kin/ecosystem/history/presenter/OrderHistoryPresenter.kt:153]: can you get rid of !! ?

File: sdk/src/main/java/com/kin/ecosystem/history/view/OrderHistoryRecyclerAdapter.java [hunk 3]
```diff
@@ -174,51 +159,50 @@ private boolean isFailed(Status status) {
...
-                case FAILED:
-                    actionText = TRANSACTION_FAILED_MSG;
-                    if(item.getError() != null) {
-                        actionText = TextUtils.isEmpty(item.getError().getMessage()) ?
-                            TRANSACTION_FAILED_MSG : item.getError().getMessage();
-                    }
-                    break;
-                default:
-                    break;
+            if (item.getStatus() == Status.FAILED) {
```
Comment (yosriz) [inline sdk/src/main/java/com/kin/ecosystem/history/view/OrderHistoryRecyclerAdapter.java:162]: no completed text anymore?

**Review by yosriz — APPROVED** [review]
LGTM

File: sdk/src/main/java/com/kin/ecosystem/history/presenter/OrderHistoryPresenter.kt (hunk 1 above)
Comment (OrenZak) [reply sdk/src/main/java/com/kin/ecosystem/history/presenter/OrderHistoryPresenter.kt]: OMG

File: sdk/src/main/java/com/kin/ecosystem/history/presenter/OrderHistoryPresenter.kt (hunk 2 above)
Comment (OrenZak) [reply sdk/src/main/java/com/kin/ecosystem/history/presenter/OrderHistoryPresenter.kt:153]: nop

File: sdk/src/main/java/com/kin/ecosystem/history/view/OrderHistoryRecyclerAdapter.java (hunk 3 above)
Comment (OrenZak) [reply sdk/src/main/java/com/kin/ecosystem/history/view/OrderHistoryRecyclerAdapter.java:162]: no, it was only for marketplace spend offers with coupon code.
And now we don't have it.

**Pull request merged**